R/eventemitters.R

```diff
@@ -164,17 +188,31 @@ Callbacks <- R6::R6Class(
       })
     },
     invoke = function(...) {
-      # Ensure that calls are invoked in the order that they were registered
-      keys <- as.character(sort(as.integer(ls(private$.callbacks)), decreasing = TRUE))
-      "!DEBUG invoke: keys = `keys`"
-      callbacks <- mget(keys, private$.callbacks)
+      callbacks <- self$get()
       "!DEBUG invoke: callback = `length(callbacks)`"
       for (callback in callbacks) {
         callback(...)
       }
     },
     count = function() {
       length(ls(private$.callbacks))
+    },
+    get = function() {
+      # Ensure that calls are invoked in the order that they were registered
+      keys <- as.character(sort(as.integer(ls(private$.callbacks)), decreasing = TRUE))
+      "!DEBUG get: keys = `keys`"
+      mget(keys, private$.callbacks)
     }
   )
 )
+
+once_function <- function(fun) {
```

Comment:
Out of curiosity, why the need of `once_function` ? For adding a specific class to `fun` ? 
I thought removing the listener was a guaranty of running it once. 🤔

Reply:
I've played a lot in node.js with the `events` module. I wanted to understand some behaviors.
More precisely, when I ran [those examples](https://nodejs.org/api/events.html#events_emitter_rawlisteners_eventname), I realized that `rawListeners` just run once. This was not the case with our current implementation.
In node.js, `rawListeners` are `listeners` wrapped in a `once wrapper` object.